### ISSUE TITLE
Question: AdvertiserTrackingEnabled is for ATTrackingManager.requestTrackingAuthorization?

### ISSUE THREAD

**Issue by isharing0726** [facebook/facebook-ios-sdk#1639]
Hi,

I'm confused about what value should I set for FBSDKSettings.AdvertiserTrackingEnabled. We're not requesting ATTrackingManager.requestTrackingAuthorization,  but we'd like to share events with FB ads for better performance.

In this case, this value should be false? or it is not related to ATTrackingManager.

**Comment by wx0165927473** [comment]
It is related to ATTrackingManager. If you don't request tracking permission prompt on iOS 14+, advertiserTrackingEnabled should be false and it will be false by default if not explicitly set.

**Comment by isharing0726** [comment]
@wx0165927473  Our FB campaigns are optimized with an App event.  if we set advertiserTrackingEnabled to false, will the campaigns receive aggregated data through SKAdnetwork? or we can't use app event optimization anymore?

**Comment by wx0165927473** [comment]
You will receive data from SKAdnetwork, however, SKAdnetwork is mainly for install event. For AEO, please check the updates in ad manager when iOS 14.5 releases.

**Comment by mattiacantalu** [comment]
Question: is it mandatory set a value after ATT prompt? I think the SDK if ATT is authorised sends to backend the IDFA, otherwise it sends just zeros. So if the scenario is to send IDFA, the SDK value can be untouched. Am I right?

**Comment by jssez28** [comment]
wx0165927473aI also need an answer to this question.

**Comment by wx0165927473** [comment]
> Question: is it mandatory set a value after ATT prompt? I think the SDK if ATT is authorised sends to backend the IDFA, otherwise it sends just zeros. So if the scenario is to send IDFA, the SDK value can be untouched. Am I right?

Yes on iOS 14.5 or later, the default value for AdvertiserTrackingEnabled is disallowed. If you don't set AdvertiserTrackingEnabled, SDK will drop the event.

**Comment by mattiacantalu** [comment]
Thanks @wx0165927473 , got it!

**Comment by xiepaul888** [comment]
On iOS 14.5 , i have set this API for true: "[FBSDKSettings setAdvertiserTrackingEnabled:YES];" but i did not  receive Launch Events, Do i have to do anything else？

**Comment by xiepaul888** [comment]
On iOS 14.5 , i have set this API for true: "[FBSDKSettings setAdvertiserTrackingEnabled:YES];" but i did not receive Launch Events , Do i have to do anything else？

**Comment by xiepaul888** [comment]
Hi ：
On iOS 14.5 , i have set this API for true: "[FBSDKSettings setAdvertiserTrackingEnabled:YES];" but i did not receive Launch Events , Do i have to do anything else？

**Comment by danmaas** [comment]
I also had a LOT of trouble getting App Install events to show up. Here is what worked:

1. set AdvertiserTrackingEnabled YES
2. *manually* log an event called `fb_mobile_first_app_launch` (for some reason the SDK doesn't seem to do this automatically, even though I have set AutoLogAppEventsEnabled YES)
3. add a parameter with key `_implicitlyLogged` and value `1` to all events. (if I don't do this, then the events do not show up in the Facebook Events Debugging tool).

With the above tweaks, I do see the correct App Install and App Launch events in my Facebook app's Insights console.

However, I am still not seeing a correct number for App Installs in the Facebook Ad Manager for my iOS 14 app install ads. I have all the SKAdNetwork features enabled, but I am only seeing a small fraction of the conversion events that I expect.

**Comment by wx0165927473** [comment]
@danmaas @xiepaul888 Could you create a new issue for better tracking? Thanks

**Comment by lukestudios** [comment]
Does anyone have clarity on AdvertiserTrackingEnabled, events with SKAdNetwork and without ATT/IDFA?

Goal: For our ad campaigns, we want to attribute app installs and optimise around select events.

We do not want to access the IDFA and show the ATT prompt. It is our understanding this can be done by using SKAdNetwork. Is this correct?

I'm struggling to find clarity around AdvertiserTrackingEnabled and SKAdNetwork. When must AdvertiserTrackingEnabled be enabled?
Is it only after receiving user consent from the ATT prompt?
Must it be enabled to use SKAdNetwork?

If AdvertiserTrackingEnabled is disabled, the documentation states events will be dropped. How can SKAdNetwork events be configured in the Event Manager if there are no events visible?
